test/conftest.py

```diff
@@ -31,11 +30,7 @@ def rootdir():
     resolved and copied as files.
     """
 
-    def copytree(src, dest):
-        shutil.copytree(src, dest, symlinks=True)
-
-    with patch.object(sphinx.testing.path.path, "copytree", copytree):
-        yield sphinx.testing.path.path(SOURCE_ROOT_DIR)
+    yield Path(SOURCE_ROOT_DIR)
```

Comment:
Using `pathlib.Path` instead of the bespoke `sphinx.testing.path` would definitely be an improvement, but it only works on Sphinx 7, and breaks on Sphinx 5 (the current minimum [here](https://github.com/melissawm/sphinx-tags/blob/main/pyproject.toml#L19)) and 6.

Reply:
Sure, it's worth a try!

If that doesn't work out, you can ignore specific warnings like this:
```py
import warnings

with warnings.catch_warnings():
    warnings.simplefilter('ignore', category=DeprecationWarning)
    # code that raises an expected DeprecationWarning to ignore
```

That will silence DeprecationWarnings for a specific block of code instead of disabling them globally, so you still see warnings from other places you aren't expecting. That's occasionally useful when testing something like this that needs to be compatible with older versions of a library.

If you can get `pathlib.Path` to work, though, that would be even better.